### PR TITLE
Add a `CITATION.cff` file

### PR DESCRIPTION
**Pull request by brandonwillard**
This PR adds a `CITATION.cff`.  See https://github.com/aesara-devs/aesara/discussions/1504#discussioncomment-6415430.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    affiliation: University of Chicago
+    orcid: 'https://orcid.org/0000-0002-0951-3289'
+  - given-names: Rémi
+    family-names: Louf
+  - given-names: Kaustubh
+    family-names: Chaudhari
+  - given-names: Ben
+    family-names: Mares
+  - given-names: Smit
+    family-names: Lunagariya
```
Comment (Smit-create) [inline CITATION.cff:24]: Should I also add my email/affiliation or it's fine?

**Review by Smit-create — DISMISSED** [review]
Thank you @brandonwillard!

**Review by rlouf — APPROVED** [review]

File: CITATION.cff (hunk 1 above)
Comment (brandonwillard) [reply CITATION.cff:24]: Yeah, feel free to add it or provide the diff/info.

Comment (Smit-create) [reply CITATION.cff:24]: Sure,
Email: `smitlunagariya.mat18@itbhu.ac.in`
Affiliation: `Indian Institute of Technology (BHU), Varanasi`

Comment (Smit-create) [inline CITATION.cff:24]:
```suggestion
    family-names: Lunagariya
    email: smitlunagariya.mat18@itbhu.ac.in
    affiliation: Indian Institute of Technology (BHU), Varanasi
```

**Review by Smit-create — APPROVED** [review]

**Pull request merged**